packages/php-wasm/compile/build-assets/php_wasm.c

```diff
@@ -91,11 +92,31 @@ EMSCRIPTEN_KEEPALIVE inline int php_pollfd_for(php_socket_t fd, int events, stru
 	return n;
 }
 
+ZEND_BEGIN_ARG_INFO_EX(arginfo_post_message_to_js, 0, 0, 2)
+    ZEND_ARG_INFO(0, type)
+    ZEND_ARG_INFO(0, data)
+ZEND_END_ARG_INFO()
 
 ZEND_BEGIN_ARG_INFO(arginfo_dl, 0)
 	ZEND_ARG_INFO(0, extension_filename)
 ZEND_END_ARG_INFO()
 
+/* {{{ proto int strcmp(string str1, string str2)
+   Binary safe string comparison */
+PHP_FUNCTION(post_message_to_js)
+{
+	char *s1, *s2;
+	int s1_len, s2_len;
+	
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "ss", &s1, &s1_len, &s2, &s2_len) == FAILURE) {
+		return;
+	}
+
+	js_module_onMessage(s1, s2);
+}
+/* }}} */
```

Comment:
this just made me realize it's trivial to explore changes to PHP with WordPress using the Playground.

we could build the Tag Processor inside PHP as an exploration and expose it via PHP itself into WordPress

```php
$p = html_tag_processor_init( $html );
while ( html_tag_processor_next( $p, $query ) ) {
	$tag_name = html_tag_processor_get_tag( $p );
}
```

Reply:
Nice, yes! Nit: I like a class better than a resource in this context. Either works, though.

Reply:
in the case of PHP it wouldn't be my choice either way. most of the extensions have class and function interfaces.